spec/rspec/mocks/reraising_eager_raises_spec.rb

```diff
@@ -0,0 +1,47 @@
+require "spec_helper"
+
+RSpec.describe "Reraising eager raises during the verify step" do
+  it "does not reraise when a double receives a message that hasn't been allowed/expected" do
+    with_unfulfilled_double do |dbl|
+      expect { dbl.foo }.to fail
+      expect { verify_all }.not_to raise_error
+    end
+  end
+
+  it "reraises when a negative expectation receives a call" do
+    with_unfulfilled_double do |dbl|
+      expect(dbl).not_to receive(:foo)
+      expect { dbl.foo }.to fail
+      expect { verify_all }.to raise_error(/expected: 0 times with any arguments/)
```

Comment:
`raise_error(regex)` here and below can instead be `fail_with(regex)` (see `spec/support/matchers.rb`).  That's a bit more descriptive and is a stronger expectation since it specifies the expectation class for you.